### ISSUE TITLE
Validator default string parsed to number

### ISSUE THREAD

**Issue by CunliangGeng** [dynaconf/dynaconf#1064]
**Describe the bug**

The default string value set in `Validator` is parsed to number, for example:
`Validator("version", is_type_of=str, default="0.30", apply_default_on_none=True),` 
The setting `version` is expected to be a string `"0.30"`, but will get a number `0.3`.  
Changing double quote in `default="0.30"` to single (`'0.30'`) or triple (`'''0.30'''` or `"""0.30"""`) still output a number `0.3`.

**To Reproduce**

1. Having the following folder structure

<details>
<summary> Project structure </summary>

```
.
├── app.py
└── settings.toml
```
</details>

2. Having the following config files:

<details>
<summary> Config files </summary>

The `settings.toml` is an empty file.

</details>

3. Having the following app code:

<details>
<summary> Code </summary>

**./app.py**
```python
from dynaconf import Dynaconf
from dynaconf import Validator


# Load config files
config = Dynaconf(settings_files=["settings.toml"])


validators = [
    Validator(
        "version", is_type_of=str, default="0.30", apply_default_on_none=True
    ),
]

config.validators.register(*validators)
config.validators.validate()


print(type(config.version))
print(config.version)
```

</details>

4. Executing under the following environment

<details>
<summary> Execution </summary>

```bash
# other commands and details?
# virtualenv activation?

$ python app.py
```

After execution, you will get an ValidationError:

```
Traceback (most recent call last):
  File "app.py", line 16, in <module>
    config.validators.validate()
  File "/opt/homebrew/anaconda3/envs/npl3.10/lib/python3.10/site-packages/dynaconf/validator.py", line 492, in validate
    validator.validate(
  File "/opt/homebrew/anaconda3/envs/npl3.10/lib/python3.10/site-packages/dynaconf/validator.py", line 216, in validate
    self._validate_items(
  File "/opt/homebrew/anaconda3/envs/npl3.10/lib/python3.10/site-packages/dynaconf/validator.py", line 340, in _validate_items
    raise ValidationError(_message, details=[(self, _message)])
dynaconf.validator.ValidationError: version must is_type_of <class 'str'> but it is 0.3 in env main
```

</details>

**Expected behavior**
The ValidationError is not expected. The value of the setting `version` is expected to be a string `"0.30"` but not a number `0.30`.

**Environment (please complete the following information):**
 - OS: MacOS (M1) 14.2.1 (23C71)
 - Dynaconf Version 3.2.4

**Comment by rochacbruno** [comment]
Working on a fix, for now the workaround is to force the cast

```python
validators = [
    Validator(
        "version", 
        is_type_of=str, 
        default="0.30", 
        apply_default_on_none=True,
        cast=str,  # <-- THIS
    ),
]
```

Ideally it would assume the type passed on `is_type_of` as the casting type

**Comment by rochacbruno** [comment]
The final fix is in draft https://github.com/dynaconf/dynaconf/pull/1066 and will be merged only on 3.3.0 because it is a bit of a breaking change if someone is relying on the opposite behavior.

I suggest you to use the provided workaround using cast=str for now.

thanks
